sdk/cosmos/azure-cosmos/docs/Metrics.md

```diff
@@ -0,0 +1,174 @@
+
+## How to enable metrics?
+
+To enable client metrics a `io.micrometer.core.instrument.MeterRegistry` needs to be passed to the `CosmosClientBuilder` when creating the singleton Cosmos client.
+
+```java
+this.client = new CosmosClientBuilder()
+    .endpoint("<Endpint>")
+    .key("<Key>")
+    .clientTelemetryConfig().clientMetrics(createConsoleLoggingMeterRegistry())
+    .buildClient();
+```
+
+A sample for a `MeterRegistry` simply logging to the Console can be create like this:
+
+```java
+private static MeterRegistry createConsoleLoggingMeterRegistry() {
+
+    final MetricRegistry dropwizardRegistry = new MetricRegistry();
+
+    final ConsoleReporter consoleReporter = ConsoleReporter
+        .forRegistry(dropwizardRegistry)
+        .convertRatesTo(TimeUnit.SECONDS)
+        .convertDurationsTo(TimeUnit.MILLISECONDS)
+        .build();
+
+    consoleReporter.start(1, TimeUnit.SECONDS);
+
+    DropwizardConfig dropwizardConfig = new DropwizardConfig() {
+
+        @Override
+        public String get(@Nullable String key) {
+            return null;
+        }
+
+        @Override
+        public String prefix() {
+            return "console";
+        }
+
+    };
+
+    final DropwizardMeterRegistry consoleLoggingRegistry = new DropwizardMeterRegistry(
+        dropwizardConfig, dropwizardRegistry, HierarchicalNameMapper.DEFAULT, Clock.SYSTEM) {
+        @Override
+        protected Double nullGaugeValue() {
+            return Double.NaN;
+        }
+
+        @Override
+        public void close() {
+            super.close();
+            consoleReporter.stop();
+            consoleReporter.close();
+        }
+    };
+
+    consoleLoggingRegistry.config().namingConvention(NamingConvention.dot);
+    return consoleLoggingRegistry;
+}
+```
+
+The micrometer.io documentation has a list with samples on how to create a `MeterRegistry` for several telemetry systems: `https://micrometer.io/docs`. You can also find an implementation via an Azure Monitor `MeterRegistry` here: [Spark sample with Azure Monitor `MeterRegistry`](https://aka.ms/azure-cosmos-spark_CosmosMetricsApplicationInsightsPlugin)
+
+
+
+## Which tags/dimensions are associated with metrics?
+
+| Tag/dimension name                   | Sample value                                                 | Description                                                  | Scope                       |
+| ------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | --------------------------- |
+| `Container`                          | `accountName / databaseName / containerName`                 | The fully qualified container name                           | operations + requests       |
```

Comment:
instead of the containerName, should we use containerRid?

Reply:
No - doesn't tell the customer anything. Metric names need to be as reasonably understandable for customers as possible (I admit that for service endpoint and service address and other rntbd metrics this is hard - but for tags we should be able to make it understandable.